4337/contracts/Safe4337Module.sol

```diff
@@ -20,28 +20,56 @@ import {ISafe} from "./interfaces/Safe.sol";
  *      - Replay protection is handled by the entry point.
  */
 contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandler {
+    /**
+     * @dev A structure used internally for computing the EIP-712 signing message for a Safe operation.
+     */
+    struct SafeOpFields {
+        bytes32 typeHash;
+        address safe;
+        uint256 nonce;
+        bytes32 initCodeHash;
+        bytes32 callDataHash;
+        uint256 callGasLimit;
+        uint256 verificationGasLimit;
+        uint256 preVerificationGas;
+        uint256 maxFeePerGas;
+        uint256 maxPriorityFeePerGas;
+        bytes32 paymasterAndDataHash;
+        uint48 validAfter;
+        uint48 validUntil;
+        address entryPoint;
+    }
+
+    /**
+     * @notice The EIP-712 type-hash for the domain separator used for verifying Safe operation signatures.
+     */
     bytes32 private constant DOMAIN_SEPARATOR_TYPEHASH = keccak256("EIP712Domain(uint256 chainId,address verifyingContract)");
```

Comment:
Would it be more gas efficient if we hardcode the hash? (might be that the optimizer handles this)

Reply:
Let me check. I think this is a separate change, so I will keep it out of this PR for now, but do a follow-up in case it helps.

Reply:
WOW - I am flabbergasted that it actually makes a difference… It is small (26 gas) but still…

Reply:
Captured in https://github.com/safe-global/safe-modules/issues/179